<?xml version="1.0"?>
<launch>

    <node pkg="love_maker_2000" type="cylinder_detector" name="cylinder_detector">

        <remap from="input" to="/love_maker_2000/voxelgrid"/>

        <param name="cylinder_normal_distance_weight" value="0.1" />  <!-- 0.1 -->
        <param name="cylinder_max_iterations" value="10000" />   <!-- 10000 -->
        <param name="cylinder_distance_threshold" value="0.09" />   <!-- 0.05 -->
        <param name="cylinder_radius_min" value="0.11" />    <!-- 0.06 -->
        <param name="cylinder_radius_max" value="0.13" />    <!-- 0.2 -->
        <param name="cylinder_inliers_threshold" value="200"/>

        <param name="plane_normal_distance_weight" value="0.1" />  <!-- 0.1 -->
        <param name="plane_max_iterations" value="100" />   <!-- 10000 -->
        <param name="plane_distance_threshold" value="0.03" />   <!-- 0.05 -->
        <param name="plane_inliers_threshold" value="100" />

        <!-- How far can we see the points in the point cloud -->
        <param name="max_point_cloud_depth" value="2.5" />

    </node>

    <!-- Node that makes the cloud point sparser -->
    <node pkg="love_maker_2000" type="voxelgrid_new" name="voxelgrid" output="screen">

        <remap from="input" to="/camera/depth/points"/>
        <remap from="output" to="/love_maker_2000/voxelgrid"/>

        <!-- Leaf size in meters -->
        <param name="leaf_size_x" value="0.03"/> <!-- 0.01 -->
        <param name="leaf_size_y" value="0.03"/> <!-- 0.01 -->
        <param name="leaf_size_z" value="0.03"/> <!-- 0.01 -->

    </node>

</launch>
